File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.go
func isVowel(c string) bool {
	if c == "a" || c == "e" || c == "i" || c == "o" || c == "u" {
		return true
	}
	return false
}

func maxVowels(s string, k int) int {
	left := 0
	right := 0
	vowelCount := 0

	for right < k {
		if isVowel(string(s[right])) {
			vowelCount++
		}
		right++
	}

	maxCount := vowelCount

	for right < len(s) {
		if isVowel(string(s[left])) {
			vowelCount--
		}

		if isVowel(string(s[right])) {
			vowelCount++
		}
        
        right++
        left++

		maxCount = max(maxCount, vowelCount)
	}

	return maxCount
}

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.java
/** 
Given a string s and an integer k, return the maximum number of vowel letters in any substring of s with length k.
Vowel letters in English are 'a', 'e', 'i', 'o', and 'u'.
**/

class Solution {


    private boolean isVowel(char c){
        if(c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u') {
            return true;
        }
        return false;
    }

    public int maxVowels(String s, int k) {
        int left = 0, right = 0;
        int vowelCount = 0;

        while(right < k) {
            if(isVowel(s.charAt(right))) {
                vowelCount++;
            }
            right++;
        }

        int maxVowels = vowelCount;
        while(right < s.length()) {
            if(isVowel(s.charAt(left++))) {
                vowelCount--;
            }


            if(isVowel(s.charAt(right++))) {
                vowelCount++;
            }

            maxVowels = Math.max(vowelCount, maxVowels);
        }
        return maxVowels;
    }
}